stream-chat-android-client/src/main/java/io/getstream/chat/android/client/socket/SocketFactory.kt

```diff
@@ -49,13 +50,21 @@ internal class SocketFactory(
     }
 
     private fun buildUserDetailJson(user: User?): String {
-        val data = mutableMapOf<String, Any>()
-        user?.let {
-            data["user_details"] = user
-            data["user_id"] = user.id
-        }
-        data["server_determines_connection_id"] = true
-        data["X-STREAM-CLIENT"] = ChatClient.instance().getVersion()
+        val user = user ?: anonymousUser()
+        val data = mapOf(
+            "user_details" to user,
+            "user_id" to user.id,
+            "server_determines_connection_id" to true,
+            "X-STREAM-CLIENT" to ChatClient.instance().getVersion()
+        )
         return parser.toJson(data)
     }
+
+    /**
+     * It doesn't matter what user id we send to the server as the anonymous
+     * user id will always be "!anon"
```

Comment:
Have you talked with Backend team about that?
If the `userId` of an anonymous user is always `!anon`, why are we using a random UUID instead of using `!anon`?

Reply:
I've inspected JS sources, and they are sending UUID, probably to indicate that any string is fine. I will replace UUID with "!anon", will be more meaningful

Reply:
Server not accepting "!anon" as the user id due to the exclamation mark, will use "anon" then

Reply:
I also added a tiny explanation that no matter what user id we send to the server we get "!anon" in return